devito/ir/equations/algorithms.py

```diff
@@ -94,13 +95,20 @@ def lower_exprs(expressions, **kwargs):
         mapper = {f: f.indexify(lshift=True, subs=dimension_map)
                   for f in retrieve_functions(expr)}
 
+        nsubs = {}
         # Handle Indexeds (from index notation)
-        for i in retrieve_indexed(expr):
+        for i in retrieve_indexed(expr, deep=True):
             f = i.function
 
             # Introduce shifting to align with the computational domain
             indices = [(a + o) for a, o in zip(i.indices, f._size_nodomain.left)]
 
+            # Indexify indices of nested functions
```

Comment:
Maybe I am beeing stupid, but can't  all of this complicated stuff be replaced by
`indices = [(lower_exprs(a) + o) for a, o in zip(i.indices, f._size_nodomain.left)]`

Reply:
I have to report that you faded me my PR and everything. Nice (recursive) catch.
SImplicity at its best!